examples/api/lib/material/date_picker/show_date_picker.1.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for basic [showDatePicker].
+
+void main() => runApp(const DatePickerApp());
+
+class DatePickerApp extends StatelessWidget {
+  const DatePickerApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: DatePickerExample(),
+    );
+  }
+}
+
+class DatePickerExample extends StatefulWidget {
+  const DatePickerExample({super.key});
+
+  @override
+  State<DatePickerExample> createState() => _DatePickerExampleState();
+}
+
+class _DatePickerExampleState extends State<DatePickerExample>{
+
+  DateTime? selectedDate;
+
+  Future<void> _selectDate(BuildContext context) async {
```

Comment:
A nice thing about stateful widgets is that they have access to the `context` outside the build method!

```suggestion
  Future<void> _selectDate() async {
```